### ISSUE TITLE
revert broken merge on categories.txt - pb applying git diff

### ISSUE THREAD

**Issue by stephanegigandet** [openfoodfacts/openfoodfacts-server#4065]
There was an issue with a merge that mistakenly removed changes to the categories.txt file

https://github.com/openfoodfacts/openfoodfacts-server/pull/3970/commits/bf0838180146eb9384cebb9f74fa4143a4702082

The categories.txt file has been modified since. I'd like to revert the merge, only for the categories.txt file, and keep the newer changes to categories.txt

I have a diff file, but I can't apply it, as it seems there's some encoding issue.

root@ns3135968:/home/off/openfoodfacts-server# git show -R bf08381 --color=never taxonomies/categories.txt > instructions
root@ns3135968:/home/off/openfoodfacts-server# more instructions 
commit bf0838180146eb9384cebb9f74fa4143a4702082
Merge: 89f60c588 92a354435
Author: Stéphane Gigandet <stephane@openfoodfacts.org>
Date:   Sat Aug 1 21:06:10 2020 +0000

    merged master, fixed conflict

diff --cc taxonomies/categories.txt
index ccbb9a1a3,ccbb9a1a3..370077ce4
--- a/taxonomies/categories.txt
+++ b/taxonomies/categories.txt
@@@ -188,6 -188,6 +188,7 @@@ fr:Biscuits pour bébé, Biscuit pour b
  hu:Kekszek csecsemőknek
  nl:Babykoekjes, Koekjes voor babies
  ro:Biscuiți pentru bebeluși, Biscuiți bebeluși
++agribalyse_food_code:en:24689
  ciqual_food_code:en:24689
  ciqual_food_name:en:Biscuit for baby
  ciqual_food_name:fr:Biscuit pour bébé
@@@ -227,6 -227,6 +228,7 @@@ ciqual_food_name:fr:Boisson infantile c
  <en:Milky cereal-based beverages
  en:Dairy cereal-based beverage for baby's breakfast
  fr:Boisson lactée pour petit-déjeuner bébé avec céréales
++agribalyse_food_code:en:13173
  ciqual_food_code:en:13173
  ciqual_food_name:en:Dairy cereal-based beverage for baby's breakfast
root@ns3135968:/home/off/openfoodfacts-server# git apply instructions 
error: unrecognized input

**Comment by stephanegigandet** [comment]
broken patch file
[categories-patch-file.txt](https://github.com/openfoodfacts/openfoodfacts-server/files/5117390/categories-patch-file.txt)

**Comment by hangy** [comment]
It should be possible to do that from the console: https://dev.to/lofiandcode/git-and-github-how-to-revert-a-single-file-dha

**Comment by stephanegigandet** [comment]
@hangy : the issue is that I don't want to revert to a specific version of the categories.txt file, as there have been 9 others commit to it since: https://github.com/openfoodfacts/openfoodfacts-server/commits/master/taxonomies/categories.txt

so either I find a way to revert the changes made in that specific commit and apply them to the current version, or we could try to revert to the version before the wrong commit, and find a way to reapply the following changes on top of it.

**Comment by VaiTon** [comment]
@stephanegigandet you can **revert** a commit and **reset** to a commit. 

When you **revert** a commit, every change of that commit gets reverted, everything added gets removed and everything removed gets added again. Then git merges the changes with current HEAD. **That's what you want to do now.**

`git revert bf08381`

When you **reset** to a commit, you just tell git to forget the commits made after that commit, so either you reset every file to that commit state, or you keep the files as they are but they are added to the index.

Also note that after resetting you have to force-push because it has modified history, something not recommended on a public branch.

**Comment by stephanegigandet** [comment]
@VaiTon Thanks! I was mistaken about the wrong commit, I was trying to revert the merge commit, but I found which individual commit removed the entries, so I reverted that one instead:

https://github.com/openfoodfacts/openfoodfacts-server/pull/4067

**Comment by Ban3** [comment]
I don't know if it's the best way of doing this, but looking at the the magit log, I started with the equivalent of:
` git revert --no-commit --edit aeeca45`
Reverting the commit but not commiting the changes. After that the changes can be modified like any other changes to the worktree. In this case I just discarded everything outside of categories and ingredients txt, and refined the changes to those a bit.
